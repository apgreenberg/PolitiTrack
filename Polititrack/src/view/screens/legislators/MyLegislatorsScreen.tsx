import React from 'react';
import { Text, View, SafeAreaView } from 'react-native';
import 'react-native-gesture-handler';
import CustomButton from '../../components/CustomButton';
import MyItemsList from '../../components/lists/MyItemsList';

/**
 * Displays a list of the user's legislators
 * @returns - the functional components representing the screen
 */
const MyLegislatorsScreen: React.FC = () => {
  return (
    <SafeAreaView>
      <View style={{ height: 90, borderBottomWidth: 3 }}>
        <CustomButton type="Find My Legislators" />
      </View>
      <MyItemsList type="legislators" />
    </SafeAreaView>
  );
};

export default MyLegislatorsScreen;
