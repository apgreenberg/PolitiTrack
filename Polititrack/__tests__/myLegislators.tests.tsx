import React from 'react';
import TestRenderer from 'react-test-renderer';
import { MockedProvider } from '@apollo/client/testing';
import MyLegislatorsScreen from '../src/view/screens/legislators/MyLegislatorsScreen';
import { useSelectorMock, useDispatchMock, userData } from './common';

/**
 * Tests the My Legislators Screen
 */
describe('My legislators tests', () => {
  beforeEach(() => {
    useSelectorMock.mockClear();
    useDispatchMock.mockClear();
  });
  test('Viewing the my legislators page', async () => {
    useSelectorMock.mockReturnValue(userData);
    const component = TestRenderer.create(
      <MockedProvider mocks={[]} addTypename={false}>
        <MyLegislatorsScreen />
      </MockedProvider>
    );
    const myLegislatorsScreen = component.toJSON();
    expect(myLegislatorsScreen).toMatchSnapshot();
    expect(useSelectorMock).toHaveBeenCalled();
  });
});
