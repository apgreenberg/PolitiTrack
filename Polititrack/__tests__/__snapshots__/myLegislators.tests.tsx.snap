// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My legislators tests Viewing the my legislators page 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View
    style={
      Object {
        "borderBottomWidth": 3,
        "height": 90,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#dae8eb",
          "borderBottomWidth": 2,
          "borderColor": "#9ca7b8",
          "borderLeftWidth": 1,
          "borderRadius": 5,
          "borderRightWidth": 1,
          "borderTopWidth": 2,
          "height": "90%",
          "justifyContent": "center",
          "opacity": 1,
          "width": "90%",
        }
      }
      testID="customButton"
    >
      <Text
        style={
          Object {
            "fontSize": 24,
            "textAlign": "center",
          }
        }
      >
        Find My Legislators
      </Text>
    </View>
  </View>
  <View>
    <View
      style={
        Object {
          "marginBottom": 180,
        }
      }
    >
      <View
        style={Object {}}
      >
        <View
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "overflow": "hidden",
                "position": "absolute",
                "right": 0,
                "top": 0,
                "zIndex": 1,
              },
              Object {
                "height": 0,
                "width": 0,
              },
            ]
          }
        >
          <View
            leftActionActivated={false}
            leftActionState={false}
            rightActionActivated={false}
            rightActionState={false}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#9ca7b8",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingLeft": 15,
              }
            }
            swipeAnimatedValue={0}
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "red",
                  "bottom": 0,
                  "justifyContent": "center",
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 75,
                }
              }
              testID="deleteSwipeButton"
            >
              <Text
                style={
                  Object {
                    "color": "#FFF",
                  }
                }
              >
                Delete
              </Text>
            </View>
          </View>
        </View>
        <View
          manipulationModes={
            Array [
              "translateX",
            ]
          }
          onLayout={[Function]}
          onMoveShouldSetResponder={[Function]}
          onMoveShouldSetResponderCapture={[Function]}
          onResponderEnd={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderReject={[Function]}
          onResponderRelease={[Function]}
          onResponderStart={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          onStartShouldSetResponderCapture={[Function]}
          style={
            Object {
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
              "zIndex": 2,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#f0f0f0",
                "borderBottomColor": "#9ca7b8",
                "borderBottomWidth": 1,
                "height": 60,
                "justifyContent": "center",
              }
            }
            testID="mainSwipeButton"
          >
            <View>
              <Text
                style={
                  Object {
                    "fontSize": 16,
                    "textAlign": "center",
                  }
                }
              >
                Scott M. Bennett
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={Object {}}
      >
        <View
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "overflow": "hidden",
                "position": "absolute",
                "right": 0,
                "top": 0,
                "zIndex": 1,
              },
              Object {
                "height": 0,
                "width": 0,
              },
            ]
          }
        >
          <View
            leftActionActivated={false}
            leftActionState={false}
            rightActionActivated={false}
            rightActionState={false}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#9ca7b8",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingLeft": 15,
              }
            }
            swipeAnimatedValue={0}
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "red",
                  "bottom": 0,
                  "justifyContent": "center",
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 75,
                }
              }
              testID="deleteSwipeButton"
            >
              <Text
                style={
                  Object {
                    "color": "#FFF",
                  }
                }
              >
                Delete
              </Text>
            </View>
          </View>
        </View>
        <View
          manipulationModes={
            Array [
              "translateX",
            ]
          }
          onLayout={[Function]}
          onMoveShouldSetResponder={[Function]}
          onMoveShouldSetResponderCapture={[Function]}
          onResponderEnd={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderReject={[Function]}
          onResponderRelease={[Function]}
          onResponderStart={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          onStartShouldSetResponderCapture={[Function]}
          style={
            Object {
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
              "zIndex": 2,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#f0f0f0",
                "borderBottomColor": "#9ca7b8",
                "borderBottomWidth": 1,
                "height": 60,
                "justifyContent": "center",
              }
            }
            testID="mainSwipeButton"
          >
            <View>
              <Text
                style={
                  Object {
                    "fontSize": 16,
                    "textAlign": "center",
                  }
                }
              >
                Carol Ammons
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
